lib/primitives/mtx.js

```diff
@@ -1122,12 +1136,31 @@ class MTX extends TX {
     assert(options, 'Options are required.');
     assert(options.changeAddress, 'Change address is required.');
 
+    // Ensure backward compatibility with the old API:
```

Comment:
Do we really want to remove old API? Other than incorrectly managing preferred inputs, was there another thing wrong with it?

As far as I can see, what we want to fix is - when `tx.clone` ing in the coinSelector, CoinSelector should use existing mtx.view as reference lookup table for these coins and don't expect them to be supplied with `coins` array that is coming from the database. That way "preferred" inputs that have coins wont get removed, but be checked against supplied coinview.